syntax = "proto2";

message request_get_truck {
    required int64 order_id = 1;
    optional string ups_account = 2;
    required int32 warehouse_id = 3;
    required int32 location_x = 4;  
    required int32 location_y = 5;  
    required int32 destination_x = 6; 
    required int32 destination_y = 7; 
}

message request_init_delivery {
    required int64 package_id = 1;
}

message request_change_destination {
    required int64 package_id = 1;
    required int32 new_destination_x = 2;
    required int32 new_destination_y = 3;
}

message AUCommands {
    optional request_get_truck get_truck = 1;
    optional request_init_delivery init_delivery = 2;
    optional request_change_destination change_destination = 3;
    optional bool disconnect = 4;
}


message response_truck_arrived {
    required int32 wh_x = 1;
    required int32 wh_y = 2;
    required int32 truck_id = 3;
    required int64 package_id = 4;
}


message response_package_delivered {
    required int64 package_id = 1;
}

message response_destination_changed {
    required int32 new_destination_x = 1;
    required int32 new_destination_y = 2;
    required int64 package_id = 3;
    required bool success = 4;
}

message UACommands {
    optional response_truck_arrived truck_arrived = 1;
    optional response_package_delivered package_delivered = 2;
    optional response_destination_changed destination_changed = 3;
    optional bool disconnect = 4;
    optional int64 world_id = 5;
}